CRM/Custom/Page/CustomDataTrait.php

```diff
@@ -35,15 +35,21 @@ trait CRM_Custom_Page_CustomDataTrait {
    *
    */
   protected function getCustomDataFieldsForEntityDisplay(string $entity, int $id): array {
-    $values = civicrm_api4($entity, 'get', [
-      'select' => [
-        'custom.*',
-      ],
-      'where' => [
-        ['id', '=', $id],
-      ],
-      'checkPermissions' => TRUE,
-    ])->single();
+    try {
+      $values = civicrm_api4($entity, 'get', [
+        'select' => [
+          'custom.*',
+        ],
+        'where' => [
+          ['id', '=', $id],
+        ],
+        'checkPermissions' => TRUE,
+      ])->single();
+    }
+    catch (CRM_Core_Exception $e) {
```

Comment:
Could this exception be made more granular? That this only catches a NotASingleResultException or something like that?

Reply:
@Sjord that would mean changing the exceptions thrown in CRM_Utils_Array::single but that might have changes required in other places. But also its probably worth catching any API exception which is generally a CRM_Core_Exception here anyway e.g. DB errors and the like.